README: Add example connect to libvirt via TLS over TCP

weizhouapache opened the pull request.

trapgate commented on README.md:
```diff
@@ -169,3 +169,90 @@ ID	Name		UUID
+        }
+
+        fmt.Println("ID\tName\t\tUUID")
+        fmt.Printf("--------------------------------------------------------\n")
```
```suggestion
        fmt.Println("--------------------------------------------------------")
```

trapgate commented on README.md:
```diff
@@ -169,3 +169,90 @@ ID	Name		UUID
+        // This dials libvirt on the local machine
+        // It connects to libvirt via TLS over TCP
+        // To connect to a remote machine, you need to have the ca/cert/key of it.
+        keyfilexml, err := ioutil.ReadFile("/etc/pki/libvirt/private/clientkey.pem")
```
```suggestion
        keyFileXML, err := ioutil.ReadFile("/etc/pki/libvirt/private/clientkey.pem")
```

trapgate commented on README.md:
```diff
@@ -169,3 +169,90 @@ ID	Name		UUID
+                log.Fatalf("%v", err)
+        }
+
+        certfilexml, err := ioutil.ReadFile("/etc/pki/libvirt/clientcert.pem")
```
```suggestion
        certFileXML, err := ioutil.ReadFile("/etc/pki/libvirt/clientcert.pem")
```

trapgate commented on README.md:
```diff
@@ -169,3 +169,90 @@ ID	Name		UUID
+                log.Fatalf("%v", err)
+        }
+
+        cafilexml, err := ioutil.ReadFile("/etc/pki/CA/cacert.pem")
```
```suggestion
        caFileXML, err := ioutil.ReadFile("/etc/pki/CA/cacert.pem")
```

trapgate commented on README.md:
```diff
@@ -169,3 +169,90 @@ ID	Name		UUID
+        if err != nil {
+                log.Fatalf("%v", err)
+        }
+        cert, err := tls.X509KeyPair([]byte(certfilexml), []byte(keyfilexml))
```
```suggestion
        cert, err := tls.X509KeyPair([]byte(certFileXML), []byte(keyFileXML))
```

trapgate commented on README.md:
```diff
@@ -169,3 +169,90 @@ ID	Name		UUID
+        }
+
+        roots := x509.NewCertPool()
+        roots.AppendCertsFromPEM([]byte(cafilexml))
```
```suggestion
        roots.AppendCertsFromPEM([]byte(caFileXML))
```

trapgate commented on README.md:
```diff
@@ -169,3 +169,90 @@ ID	Name		UUID
+        }
+        fmt.Println("Version:", v)
+
+        domains, err := l.Domains()
```
```suggestion
        // Return both running and stopped VMs
        flags := libvirt.ConnectListDomainsActive | libvirt.ConnectListDomainsInactive
        domains, _, err := l.ConnectListAllDomains(1, flags)
```
Because `Domains()` is deprecated.

trapgate reviewed (approved):
Thank you for this example! I think this will be very helpful. I made a few small changes to bring the naming in line with typical go names, and switched one call to the non-deprecated version.

weizhouapache commented:
> Thank you for this example! I think this will be very helpful. I made a few small changes to bring the naming in line with typical go names, and switched one call to the non-deprecated version.

@trapgate thanks !
Sorry I removed the tls port in the example by mistake. I have created a pr to add it #119

trapgate commented:
Got it; thanks again!

MattIPv4 commented:
Hey, @weizhouapache - Thanks a ton for this sweet PR! 😄

Would you please shoot me an email when you get a chance?
mcowley at digitalocean dot com 🎉
